### PR TITLE
Use string views to avoid copying strings

### LINKED ISSUE #4064: Change return type of getScriptBaseName to string_view
`getScriptBaseName` functions exist in event types (`Actions`, `Weapons`, etc) and return a constant string referring to the name of the class. It is possible to, instead of returning `std::string`, return `std::string_view` which is more lightweight and avoids copying data on every call (a `string` has to be created, where a `string_view` will just be a pointer to static data).

Tip: just search for `getScriptBaseName`, it's very likely that a search and replace is enough. Optionally, move all definitions to cpp files.

### PR DESCRIPTION
**Pull request by ranisalt**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Avoids some string copies by using `string_view` to wrap existing strings instead of copying to a new string instance. Most notably, avoids copying strings from database column names and query results, and avoids copying strings when reading prop streams.

Additionally, throws in a bit of code modernization by replacing output parameters with returning pairs and destructuring assignments.

**Issues addressed:**
Closes #4064 

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: src/baseevents.cpp [hunk 1]
```diff
@@ -17,7 +17,7 @@ bool BaseEvents::loadFromXml()
 		return false;
 	}
 
-	std::string scriptsName = getScriptBaseName();
+	std::string scriptsName{getScriptBaseName()};
 	std::string basePath = "data/" + scriptsName + "/";
```
Comment (ranisalt) [inline src/baseevents.cpp:21]: Thinking about using `std::filesystem::path` here

**Review by DSpeichert — DISMISSED** [review]

**Review by EPuncker — DISMISSED** [review]

**Comment by ranisalt** [comment]
Any server owner with a relatively large server willing to try and share the results? Can't properly measure with local benchmarks :sweat_smile:

**Review by EPuncker — DISMISSED** [review]

**Comment by yamaken93** [comment]
If we rely on fmt instead of c++ 20 formatting api, we can use fmt::memory_buffer buffer + fmt::format_to + string_view to prevent copies in queries and other stuff. Here a really small optimization based on this concept:

```
	fmt::memory_buffer buffer;
	const ItemType& it = Item::items[item->getID()];
	if (!it.showCount) {
		fmt::format_to(std::back_inserter(buffer), FMT_COMPILE("Using one of {:s}..."), item->getName());
	} else if (count == 1) {
		fmt::format_to(std::back_inserter(buffer), FMT_COMPILE("Using the last {:s}..."), item->getName());
	} else {
		fmt::format_to(std::back_inserter(buffer), FMT_COMPILE("Using one of {:d} {:s}..."), count, item->getPluralName());
	}
	player->sendTextMessage(MESSAGE_HOTKEY_PRESSED, std::string_view(buffer.data(), buffer.size()));
```

This is a "avoid" copying the final string at all costs approach. The fmt::memory_buffer is a 500 stack allocated buffer that is used internally by fmt::format function and here we are using it directly to avoid the copy (and extra allocations if the final string is bigger than 500 bytes) from the stack to a new allocated in the heap string which would happen if we used fmt::format function.

**Comment by ranisalt** [comment]
@yamaken93 this is a good idea in general, but I would be careful not to overuse it

**Comment by yamaken93** [comment]
> @yamaken93 this is a good idea in general, but I would be careful not to overuse it

I agree, its a "war" on allocation + copy idea. Can be used for strings retrieved from the lua API(when lua calls TFS C functions), strings from the protocol, etc.

**Review by ramon-bernardo — DISMISSED** [review]

**Review by EPuncker — APPROVED** [review]

**Comment by ranisalt** [comment]
@EPuncker @ramon-bernardo may I get an updated review? Rebased and fixed a conflict

**Comment by ranisalt** [comment]
Rebased against master

**Review by EPuncker — APPROVED** [review]

**Review by MillhioreBT — APPROVED** [review]

**Pull request merged**